waku/waku_archive/archive.nim

```diff
@@ -212,21 +213,29 @@ proc loopApplyRetentionPolicy*(w: WakuArchive):
 
   return ok()
 
-proc reportStoredMessagesMetric*(w: WakuArchive):
-                                 Future[Result[void, string]] {.async.} =
+# Metrics reporting
+const WakuArchiveDefaultMetricsReportInterval* = chronos.minutes(1)
+
+proc loopReportStoredMessagesMetric*(w: WakuArchive):
+                                     Future[Result[void, string]] {.async.} =
   if w.driver.isNil():
-    return err("driver is Nil in reportStoredMessagesMetric")
+    return err("driver is Nil in loopReportStoredMessagesMetric")
 
-  let resCount = await w.driver.getMessagesCount()
-  if resCount.isErr():
-    return err("failed to get messages count: " & resCount.error)
+  while true:
```

Comment:
What's the exist condition of this loop (if it has one)?

Reply:
Seems that there's no exit condition so it collects the metrics every `WakuArchiveDefaultMetricsReportInterval` ms

Reply:
Good question and I agree that it doesn't sound natural. I tried to bring my five cents in https://github.com/waku-org/nwaku/pull/2287#discussion_r1428173349